### PR TITLE
Add indent option to dump/dumps

### PR DESCRIPTION
**Pull request by wimglenn**
Alternative to https://github.com/hukkin/tomli-w/pull/27 (keeps backwards compatibility).

Allows using `tomli_w.dumps(data, indent=2)` similar like stdlib `json.dumps`.

### CODE REVIEW

**Comment by wimglenn** [comment]
@hukkin  CI failures are not related to the changes

File: src/tomli_w/_writer.py [hunk 1]
```diff
@@ -27,22 +26,29 @@
...
 def dump(
-    __obj: Mapping[str, Any], __fp: IO[bytes], *, multiline_strings: bool = False
+    __obj: Mapping[str, Any],
+    __fp: IO[bytes],
+    *,
+    multiline_strings: bool = False,
+    indent: int = 4,
 ) -> None:
-    ctx = Context(multiline_strings, {})
+    ctx = Context(multiline_strings, {}, indent)
```
Comment (hukkin) [inline src/tomli_w/_writer.py:35]:
```suggestion
    ctx = Context(multiline_strings, {}, " " * indent)
```

File: src/tomli_w/_writer.py [hunk 2]
```diff
@@ -27,22 +26,29 @@
...
     for chunk in gen_table_chunks(__obj, ctx, name=""):
         __fp.write(chunk.encode())
 
 
-def dumps(__obj: Mapping[str, Any], *, multiline_strings: bool = False) -> str:
-    ctx = Context(multiline_strings, {})
+def dumps(
+    __obj: Mapping[str, Any], *, multiline_strings: bool = False, indent: int = 4
+) -> str:
+    ctx = Context(multiline_strings, {}, indent)
```
Comment (hukkin) [inline src/tomli_w/_writer.py:43]:
```suggestion
    ctx = Context(multiline_strings, {}, " " * indent)
```

File: src/tomli_w/_writer.py [hunk 3]
```diff
@@ -27,22 +26,29 @@
...
     return "".join(gen_table_chunks(__obj, ctx, name=""))
 
 
 class Context(NamedTuple):
     allow_multiline: bool
     # cache rendered inline tables (mapping from object id to rendered inline table)
     inline_table_cache: dict[int, str]
+    indent: int = 4
```
Comment (hukkin) [inline src/tomli_w/_writer.py:51]:
```suggestion
    indent_str: str
```

File: src/tomli_w/_writer.py [hunk 4]
```diff
@@ -136,8 +142,8 @@ def format_inline_table(obj: Mapping, ctx: Context) -> str:
 def format_inline_array(obj: tuple | list, ctx: Context, nest_level: int) -> str:
     if not obj:
         return "[]"
-    item_indent = ARRAY_INDENT * (1 + nest_level)
-    closing_bracket_indent = ARRAY_INDENT * nest_level
+    item_indent = " " * ctx.indent * (1 + nest_level)
```
Comment (hukkin) [inline src/tomli_w/_writer.py:145]:
```suggestion
    item_indent = ctx.indent_str * (1 + nest_level)
```

File: src/tomli_w/_writer.py [hunk 5]
```diff
@@ -136,8 +142,8 @@ def format_inline_table(obj: Mapping, ctx: Context) -> str:
...
+    closing_bracket_indent = " " * ctx.indent * nest_level
```
Comment (hukkin) [inline src/tomli_w/_writer.py:146]:
```suggestion
    closing_bracket_indent = ctx.indent_str * nest_level
```

File: src/tomli_w/_writer.py [hunk 6]
```diff
@@ -197,5 +203,5 @@ def is_aot(obj: Any) -> bool:
 def is_suitable_inline_table(obj: Mapping, ctx: Context) -> bool:
     """Use heuristics to decide if the inline-style representation is a good
     choice for a given table."""
-    rendered_inline = f"{ARRAY_INDENT}{format_inline_table(obj, ctx)},"
+    rendered_inline = f"{' ' * ctx.indent}{format_inline_table(obj, ctx)},"
```
Comment (hukkin) [inline src/tomli_w/_writer.py:206]:
```suggestion
    rendered_inline = f"{ctx.indent_str}{format_inline_table(obj, ctx)},"
```

**Comment by hukkin** [comment]
Thanks!

**Pull request merged**